File: frontend/src/types/rental.ts
export interface Rental {
  id: string;
  address: string;
  bedrooms: number;
  bathrooms: number;
  rent: number;
  status: string;
  notes: string | null;
  createdAt: string;
  updatedAt: string | null;
}

export interface RentalCreate {
  address: string;
  bedrooms: number;
  bathrooms: number;
  rent: number;
  status?: string;
  notes?: string | null;
}

export interface RentalUpdate {
  address?: string;
  bedrooms?: number;
  bathrooms?: number;
  rent?: number;
  status?: string;
  notes?: string | null;
}

File: src/HouseRental.Api/Models/Rental.cs
namespace HouseRental.Api.Models;

public class Rental
{
    public Guid Id { get; set; }
    public string Address { get; set; } = string.Empty;
    public int Bedrooms { get; set; }
    public decimal Bathrooms { get; set; }
    public decimal Rent { get; set; }
    public string Status { get; set; } = "available"; // available | occupied | maintenance
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}
